overlord/snapstate/snapmgr.go

```diff
@@ -130,11 +131,15 @@ func (snapst *SnapState) SetType(typ snap.Type) {
 
 // CurrentSideInfo returns the side info for the current revision in the snap revision sequence if there is one.
 func (snapst *SnapState) CurrentSideInfo() *snap.SideInfo {
-	n := len(snapst.Sequence)
-	if n == 0 {
+	if snapst.Current.Unset() {
```

Comment:
I don't remember if we have unit tests for CurrentSideInfo alone? now that is a bit indirect we probably want them